### DISCUSSION TITLE
支持对特定课程设置特定的小组件自动隐藏方式

### DISCUSSION THREAD

**Discussion by Yersmagit** [Class-Widgets/Class-Widgets#301] [Feature]
### 检查清单

- [X] 我已更新到最新版 ![最新的版本号](https://img.shields.io/github/v/release/RinLit-233-shiroko/Class-Widgets?include_releases&style=flat-square&label=) ，并看过[最新提交](https://github.com/RinLit-233-shiroko/Class-Widgets/commits/main/)，确认我想要的功能还没有实现。
- [X] 我已在 [Issues](https://github.com/RinLit-233-shiroko/Class-Widgets/issues?q=label%3ABug) 中检索，确认这一 Bug 未被提交过。
- [X] 我认为我想要的功能是用户广泛需要的，插件不能替代的，且没有超出 **软件本来** 的开发目标，而非 **添加与课表及教学辅助无关的内容** 。
- [X] 我已知晓由于开发者时间精力有限，我提交的功能请求可能需要比较久的时间才能实现。

### 背景与动机

新版本新增了小浮窗，它看起来很不错，尤其是在自习课放 Word 或 PPT 的时候。但也并不是每节课都需要这个小浮窗，它也会对一些老师上课产生干扰。

### 想要实现或优化的功能

希望软件可以让用户对特定的课程自定义特定的小组件 **隐藏方式** 和 **自动隐藏方式** 。
例如，我可以让语文课时，软件小组件使用“全部隐藏”的隐藏方式，自动隐藏方式则设定为“无”；而在晚自习、心理课、午休时，小组件使用“最小化为浮窗”的隐藏方式，自动隐藏方式则设定为“窗口最大化时自动隐藏”。
这似乎是一个超级超级艰巨的任务，不知道作者有木有精力专门搞一个这样的功能...

### 最后一步

- [X] 我认为上述的描述已经足以详细，以便开发人员理解您想要的功能。如果我的 issue 没有按照上述的要求填写，可能会被无条件关闭。

**Comment by MKStoler4096** [comment]
小组件必须要支持热重载才行吧……

**Comment by RinLit-233-shiroko** [comment]
> 小组件必须要支持热重载才行吧……

已经支持热重载了

**Comment by RinLit-233-shiroko** [comment]
规则集吗，或许能用插件实现
